drizzlepac/hlautils/catalog_utils.py

```diff
@@ -625,173 +649,169 @@ def transform_list_xy_to_ra_dec(self, list_of_x, list_of_y, drizzled_image):
-            # Note: SExtractor has "connectivity=8" which is the default for this function
-            self.sources = detect_sources(imgarr, threshold, npixels=self.size_source_box, filter_kernel=kernel)
-            self.kernel = kernel  # for use in measure_sources()
-            self.total_source_cat = source_properties(imgarr_bkgsub, self.sources, background=bkg.background,
-                                                      filter_kernel=kernel, wcs=self.image.imgwcs)
+            log.info("SEGMENT. Detecting sources in total image product.")
+            # Note: SExtractor has "connectivity=8" which is the default for detect_sources().
+            self.segm_img = detect_sources(imgarr_bkgsub, threshold, npixels=self._size_source_box, filter_kernel=self.image.kernel)
 
-        # if processing filter product, use sources identified by parent total drizzle product identify_sources() run
-        if self.tp_sources:
+            try:
+                # Deblending is a combination of multi-thresholding and watershed
+                # segmentation. Sextractor uses a multi-thresholding technique.
+                # npixels = number of connected pixels in source
+                # npixels and filter_kernel should match those used by detect_sources()
+                segm_deblended_img = deblend_sources(imgarr_bkgsub, self.segm_img, npixels=self._size_source_box,
+                                                     filter_kernel=self.image.kernel, nlevels=self._nlevels, contrast=self._contrast)
+
+                # The deblending was successful, so just copy the deblended sources back to the sources attribute.
+                self.segm_img = copy.deepcopy(segm_deblended_img)
+            except Exception as x_cept:
+                log.warning("SEGMENT. Deblending the sources in image {} was not successful: {}.".format(self.imgname, x_cept))
+                log.warning("SEGMENT. Processing can continue with the non-deblended sources, but the user should\n"
+                            "check the output catalog for issues.")
+
+            # Regardless of whether or not deblending worked, this variable can be reset to None
+            segm_deblended_img = None
+
+            # Clean up segments that are near or partially overlap the border of the image and relabel
+            # the segments to be sequential
+            self.segm_img.remove_border_labels(self._border, partial_overlap=True, relabel=True)
+
+            # The total product catalog consists only of X/Y and RA/Dec coordinates for the detected sources
+            # in the total drizzled image.  All the actual measurements are done on the filtered drizzled
+            # images using the coordinates determined from the total drizzled image.
+            self.source_cat = source_properties(imgarr_bkgsub, self.segm_img, background=self.image.bkg_background_ra,
+                                                filter_kernel=self.image.kernel, wcs=self.image.imgwcs)
```

Comment:
This operation seems unnecessary, since the same operation gets performed in 'measure_sources'.  Is there a reason 'self.source_cat' needs to be defined at this stage of processing, or can it actually wait until 'measure_sources' is called?  If it can wait, it would remove some unnecessary operations.

Reply:
This was done here to support the following debug block which generates data used for ds9 overlay.  I will move this code into the debug block for now.  In a follow on Issue/PR for "clean up" this code will move into the write method to keep all the file generation code in one place.

Reply:
CORRECTION.  The self.source_cat=source_properties()...) does need to be done here in order to get the centroid coordinates for the sources in the white light image.  The "total" catalogs (*.ecsv file) are written before the source properties are derived for the filter images.